lib/init.js

```diff
@@ -122,8 +123,10 @@ async function init (config = {}) {
       }
       logger.debug('init with openwhisk credentials')
       return CosmosStateStore.init(await wrapTVMRequest(tvm.getAzureCosmosCredentials()))
```

Comment:
(non-blocking for this PR) Do we still want to support TVM cosmos credentials in the new state lib major? I would be in favor of keeping the Cosmos provider only for BYO credentials.

Reply:
Makes sense to ditch more the generic interface but one of the considerations is, for people to easily migrate their code if they have the same api surface. I think we can ditch the generic interface to support multiple implementations (and thus drop CosmosDB) but keep the API the same.

Reply:
Yes good point, API would be the same where possible but some things related to limits and input validation will change, so this will have to be a breaking change

Reply:
done